### PR TITLE
[22.05] awscli2: 2.5.6 -> 2.7.9

### PR DESCRIPTION
**Pull request by colin-arnott-xero**
###### Description of changes

- Bump version
- Backport to fix Helm auth problem
- Fixes broken backport from #177953
- Supercedes #177949

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by colin-arnott-xero** [comment]
@bryanasdev000: it looks like we both missed the failed backport in #177953, so I am just grabbing all the intervening commits and backporting them (which does apply cleanly). Happy to squash them if you would rather this be a single atomic change.

**Comment by bryanasdev000** [comment]
> @bryanasdev000: it looks like we both missed the failed backport in #177953, so I am just grabbing all the intervening commits and backporting them all (which does apply cleanly). Happy to squash them if you would rather this be a single atomic change.

Well crapy, thanks for noticing it, I'm going to check Postgres as well, they all are critical to backport.

I have no problem to make multiples commit, but this will be needed to be accepted by someone with merge superpowers :P 

For future reference https://github.com/NixOS/nixpkgs/pull/177953#issuecomment-1158477365

File: pkgs/tools/admin/awscli2/default.nix [hunk 1]
```diff
@@ -67,6 +67,7 @@ with py.pkgs; buildPythonApplication rec {
   postPatch = ''
     substituteInPlace setup.cfg \
       --replace "colorama>=0.2.5,<0.4.4" "colorama" \
+      --replace "cryptography>=3.3.2,<37.0.0" "cryptography" \
```
Comment (bryanasdev000) [inline pkgs/tools/admin/awscli2/default.nix:70]: This maybe not needed, not sure the actual status in 22.05 branch, but if it compiles, it compiles!

**Review by bryanasdev000 — APPROVED** [review]
LGTM, will wait to see if borg is happy to.

Thanks!

File: pkgs/tools/admin/awscli2/default.nix [hunk 2]
```diff
@@ -67,6 +67,7 @@ with py.pkgs; buildPythonApplication rec {
   postPatch = ''
     substituteInPlace setup.cfg \
```
Comment (thiagokokada) [inline pkgs/tools/admin/awscli2/default.nix:68]: This looks like a good use case for `pythonRelaxDeps`. Maybe in the next bump on `master`.

**Review by thiagokokada — APPROVED** [review]

**Pull request merged**